packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
+      const result = await this.getModelAggregator(model, obj).group({ 
+        _id: `$${fieldKey}`,
+      });
+
+      return this.preProcessGroupByData({
+        result,
+        fieldKey,
+        filters: obj,
+        modelName: name,
+      });
+    }, () => true);
+  },
+
+  /**
+   * This method is the entry point to the GraphQL's Aggregation.
+   * It takes as param the model and its fields and it'll create the aggregation types and resolver to it
+   * Example:
+   *  type User {
+   *     username: String,
+   *     age: Int,
+   *  }
+   * 
+   * It'll create
+   *  type UserConnection {
+   *    values: [User],
+   *    groupBy: UserGroupBy,
+   *    aggreate: UserAggregate
+   *  }
+   * 
+   *  type UserAggregate {
+   *     count: Int
+   *     sum: UserAggregateSum
+   *     avg: UserAggregateAvg
+   *  }
+   * 
+   *  type UserAggregateSum {
+   *     age: Float
+   *  }
+   * 
+   *  type UserGroupBy {
+   *     username: [UserConnectionUsername]
+   *     age: [UserConnectionAge]
+   *  }
+   * 
+   *  type UserConnectionUsername {
+   *    key: String
+   *    connection: UserConnection
+   *  }
+   * 
+   *  type UserConnectionAge {
+   *    key: Int
+   *    connection: UserConnection
+   *  }
+   *
+   */
+  formatModelConnectionsGQL: function(fields, model, name, modelResolver) {
+    const { globalId } = model;
+
+    const connectionGlobalId = `${globalId}Connection`;
+    const aggregatorFormat = this.formatConnectionAggregator(fields, model, name);
+    const groupByFormat = this.formatConnectionGroupBy(fields, model, name);
+    const connectionFields = {
+      values: `[${globalId}]`,
+      groupBy: `${globalId}GroupBy`,
+      aggregate: `${globalId}Aggregator`,
+    };
+    
+    let modelConnectionTypes = `type ${connectionGlobalId} {${this.formatGQL(connectionFields)}}\n\n`;
+    if (aggregatorFormat) {
+      modelConnectionTypes += aggregatorFormat.type;
+    }
+    modelConnectionTypes += groupByFormat.type;
+
+    return {
+      globalId: connectionGlobalId,
+      type: modelConnectionTypes,
+      query: {
+        [`${pluralize.plural(name)}Connection(sort: String, limit: Int, start: Int, where: JSON)`]: connectionGlobalId,
+      },
+      resolver: {
+        Query: {
+          [`${pluralize.plural(name)}Connection`]: (obj, options, context) => { // eslint-disable-line no-unused-vars
+            const params = Object.assign(
+              {},
+              this.convertToParams(_.omit(options, 'where')),
+              options.where
+            );
+            return strapi.utils.models.convertParams(name, params);
+          }
+        },
+        [connectionGlobalId]: {
+          values: (obj, option, context) => {
+            // Object here contains the key/value of the field that has been grouped-by
+            // for instance obj = { where: { country: 'USA' } } so the values here needs to be filtered according to the parent value
+            return modelResolver(obj, obj, context);
```

Comment:
Bug detected here
The second `obj` must be replace by `option`

Reply:
it's not a bug, I've added a comment to explain why I did it. this is used to pass the key's value that has been grouped-by. I might need to add an explicit example to demonstrate how it works

Reply:
Ok but if I keep the double `obj`, the `groupBy` and `values` are not working, just the aggregate count.

Moreover, it generate me an error with the `convertParams` function:
`formattedValue.split is not a function` (it's not your fault but it's linked because it's pass both query filter)

@Aurelsicoko @lauriejim 
For fix this related bug, inside strapi-utils/models.js I change `} else if (key === '_sort') {` by `} else if (key === '_sort' && formattedValue) {`

Reply:
Oh really, I'll double check when I finish my work.
For the error that you faced is due to the fact we weren't correctly directing numeric values, I submitted a PR to fix it and it's merged now: https://github.com/strapi/strapi/pull/1345